lte/cloud/go/services/lte/analytics/calculations/state.go

```diff
@@ -149,24 +146,39 @@ func (x *UserMetricsCalculation) Calculate(prometheusClient query_api.Prometheus
 		}
 		users := make(map[string]struct{})
 		var exists = struct{}{}
-		var activeSessionsPerGateway map[string]float64
-		var activeSessionsPerAPN map[string]float64
+		activeSessionsPerAPN := make(map[string]float64)
 		for stateID, st := range states {
-			reportedSt := st.ReportedState.(*map[string]string)
-			sessionState, _ := (*reportedSt)["lifecycle_state"]
-			deviceID, _ := (*reportedSt)["device_id"]
-			apnID, _ := (*reportedSt)["apn"]
-			users[stateID.DeviceID] = exists
-			if sessionState == "SESSION_ACTIVE" && deviceID != "" && apnID != "" {
-				activeSessionsPerGateway[deviceID]++
-				activeSessionsPerAPN[apnID]++
+			if st.ReportedState != nil {
+				reportedSt, ok := st.ReportedState.(*state.ArbitraryJSON)
```

Comment:
This is not how we should be doing this. If we're gonna parse `subscriber_state` state type, then we need to coerce it to a concrete type rather than treating it as arbitrary json then attempting to access its fields procedurally. Please fix in a subsequent pr